### PR TITLE
fix typeform authenticator loop

### LINKED ISSUE #8410: Source Typeform improve check connection method
The method used to check connection is not improved for large accounts because make a full refresh and then after check if the `form_id` exists.

https://github.com/airbytehq/airbyte/blob/b424c1a0e78141fcaef61ba09caf9541772ea347/airbyte-integrations/connectors/source-typeform/source_typeform/source.py#L201-L214

I think should be more easy to validate if `form_id` exist, if failed raise the error:
```python
for form in form_ids:
   try:
       url = f"{TypeformStream.url_base/forms/{form_id}/response"
       auth_headers = {"Authorization": f"Bearer {config['token']}"}
       session = requests.get(url, headers=auth_headers)
       session.raise_status()
   except:
       return False, "The Form Id informed doesn't exist"
return True, None
```

### PR DESCRIPTION
**Pull request by brebuanirello-equinix**
## What
Our connection with typeform doesn't work with the current version 0.1.2, the method used to check connection is not improved for large accounts because make a full refresh and then after check if the form_id exists.

issue related:
https://github.com/airbytehq/airbyte/issues/8410

## How
Removing the loop in the full refresh forms, because the API does not respond.

#### community Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py [hunk 1]
```diff
@@ -204,12 +204,16 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
...
-            return True, None
+            for form in form_ids:
+                try:
+                    url = f"{TypeformStream.url_base}/forms/{form}/response"
+                    auth_headers = {"Authorization": f"Bearer {config['token']}"}
+                    session = requests.get(url, headers=auth_headers)
+                    session.raise_status()
+                except:
+                    return False, f"Cannot find forms with ID: {form}. Please make sure they are valid form IDs and try again."
+                return True, None
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-typeform/source_typeform/source.py:215]:
```suggestion
            for form in form_ids:
                try:
                    url = f"{TypeformStream.url_base}/forms/{form}/response"
                    auth_headers = {"Authorization": f"Bearer {config['token']}"}
                    session = requests.get(url, headers=auth_headers)
                    session.raise_status()
                except:
                    return False, f"Cannot find forms with ID: {form}. Please make sure they are valid form IDs and try again."
            return True, None
```
The return must happens only after the for-loop is finished. Overall looks good.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small change.

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def request_params(
         params["page"] = next_page_token or 1
         return params
 
-
+# same loop logic
 class TrimFormsMixin:
     def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
         form_ids = self.config.get("form_ids", [])
```
Comment (bruno-nirello) [inline airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: If user pass empty form list will trigger full refresh that will list all forms again.

Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: we can remove this logic, or put required form id in frontend and change the logic.

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: Here is necessary this logic.

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py [hunk 3]
```diff
@@ -204,12 +204,16 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
...
-            if form_ids:
-                return False, f"Cannot find forms with IDs: {form_ids}. Please make sure they are valid form IDs and try again."
+            for form in form_ids:
+                try:
+                    url = f"{TypeformStream.url_base}/forms/{form}/response"
+                    auth_headers = {"Authorization": f"Bearer {config['token']}"}
+                    session = requests.get(url, headers=auth_headers)
+                    session.raise_status()
+                except:
+                    return False, f"Cannot find forms with ID: {form}. Please make sure they are valid form IDs and try again."
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-typeform/source_typeform/source.py:214]: If the form_ids is empty the code won't check any endpoint, I think should be add a if form_ids else logic here.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
small comments, almost done 👍

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py (hunk 2 above)
Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: sorry, this is the lines that i don't agree with

https://github.com/airbytehq/airbyte/pull/8466/files#diff-b564154cf70574820f36c0953c3ae6943851423f132e7ed11ab8b5c2db32a14fR90-R92

Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: sorry, those are the lines i don't agree with
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-typeform/source_typeform/source.py#L90-L92

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py (hunk 3 above)
Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:214]: i think if the forms are empty they will go inside this part here

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-typeform/source_typeform/source.py#L90-L92

Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:214]: indeed

**Comment by bruno-nirello** [comment]
remaked to include empty form list

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py (hunk 3 above)
Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py]: remaked to include empty form list

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py (hunk 2 above)
Comment (bruno-nirello) [reply airbyte-integrations/connectors/source-typeform/source_typeform/source.py:86]: but i'ts necessary if someone want to get all forms and don't provide a list of forms

File: airbyte-integrations/connectors/source-typeform/source_typeform/source.py [hunk 4]
```diff
@@ -202,14 +202,27 @@ class SourceTypeform(AbstractSource):
...
-            if form_ids:
-                return False, f"Cannot find forms with IDs: {form_ids}. Please make sure they are valid form IDs and try again."
-            return True, None
+            try:    
+                url = f"{TypeformStream.url_base}/me"
+                auth_headers = {"Authorization": f"Bearer {config['token']}"}
+                session = requests.get(url, headers=auth_headers)
+                session.raise_for_status()
+            except requests.exceptions.BaseHTTPError as e:
+                f"Cannot authenticate, please verify token."
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-typeform/source_typeform/source.py:212]:
```suggestion
                return False, "Cannot authenticate, please verify token."
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
small change, after that we can merge bump the connector version and publish it.

**Review by marcosmarxm — APPROVED** [review]
thanks @bruno-nirello

**Pull request merged**